### ISSUE TITLE
Table 10: Detailed results on ImageNet-C at severity level 5 regarding accuracy (%).

### ISSUE THREAD

**Issue by CCJing14** [oyt9306/Decorruptor#5]
Could you provide the code for table10, please? I would greatly appreciate it if you could.

**Comment by CCJing14** [comment]
from __future__ import print_function

import argparse

import numpy as np
import torch
import torch.nn as nn
import torch.optim as optim
import torch.backends.cudnn as cudnn
cudnn.benchmark = True
from PIL import Image
from tqdm import tqdm
from utils.adapt_helpers import adapt_single, test_single, adapt_single_select,adapt_single_diffusion,test_ensemble,test_ensemble_debug,test_highest,test_voting
from utils.third_party import aug
from utils.train_helpers import build_model, prepare_test_data, prepare_test_imagenet100, tr_transforms, te_transforms, te_transforms_inc, common_corruptions
import torch
from torch.utils.data import DataLoader, Dataset
import numpy as np
import models.Res as Resnet
import time
import os
from pipeline.deccoruptor_lcm_pipe import IP2PLatentConsistencyModelPipeline
from transformers import CLIPTokenizer, CLIPTextModel, CLIPImageProcessor
from diffusers import AutoencoderKL, UNet2DConditionModel, LCMScheduler
from PIL import Image, ImageOps
import torchvision.transforms as transforms
import torchvision
from glob import glob 
from natsort import natsorted 

parser = argparse.ArgumentParser()
parser.add_argument('--dataroot', default="/workspace/")
parser.add_argument('--use_rvt', action='store_true')
parser.add_argument('--use_resnext', action='store_true')
parser.add_argument('--level', default=5, type=int)
parser.add_argument('--corruption', default='gaussian_noise')
parser.add_argument('--dataset', default='imagenet')
parser.add_argument('--resume',  default="/root/memo/imagenet-exps/results/imagenet_rn50")
parser.add_argument('--batch_size', default=1, type=int)
parser.add_argument('--prior_strength', default=16, type=int)
parser.add_argument('--optimizer', default='sgd')
parser.add_argument('--lr', default=0.00025, type=float)
parser.add_argument('--weight_decay', default=5e-4, type=float)
parser.add_argument('--niter', default=1, type=int)
parser.add_argument('--ifshuffle',  action='store_true', help='Set flag to True')
parser.add_argument('--workers', default=8, type=int, metavar='N',help='number of data loading workers (default: 4 )')
parser.add_argument('--method', default='adapt_single_selects_1_3',help='adapt_single, adapt_single_selects')
parser.add_argument('--ifdebug',  action='store_true', help='Set flag to True')
args = parser.parse_args()

net = build_model(args)

if args.dataset == 'imagenet':
    teset, testloader = prepare_test_data(args, use_transforms=False)
if args.dataset == 'imagenet100':
    teset, _ = prepare_test_imagenet100 (args)

print(f'Resuming from {args.resume}...')
ckpt = torch.load(f'{args.resume}/ckpt.pth')

print('Running...')
correct = 0

total = 0.0
start = time.time()
model_id = "/root/.cache/huggingface/hub/Anonymous-12/DeCorruptor-CM"
scheduler = LCMScheduler.from_pretrained(model_id, subfolder="scheduler")
pipe = IP2PLatentConsistencyModelPipeline.from_pretrained(model_id,
                                            torch_dtype=torch.float16, 
                                            scheduler=scheduler,
                                            use_safetensors=True, 
                                            output_type="tensor",
                                            safety_checker=None)
pipe = pipe.to("cuda")

generator = torch.Generator(device="cuda").manual_seed(1)
idx=0
for data in tqdm(testloader):
    idx += 1

    images, labels = data
    images, labels = images.cuda(), labels.cuda()

    out_image = pipe(prompt=['Clean the image'], 
            image=[images],
            num_images_per_prompt=1,
            num_inference_steps=4, 
            generator=generator,
            output_type="tensor",
            guidance_scale=7.5).images
    inputs = torch.tensor(out_image).permute(0, 3, 1, 2).cuda()

    if args.ifdebug==True:
        cor, conf, ent, pred = test_ensemble(net, inputs, labels, args.corruption, args.prior_strength,args.ifdebug,idx)
    cor, conf, ent, pred = test_ensemble(net,images, labels, args.corruption, args.prior_strength,args.ifdebug)
    
    correct += cor
    
    total += float(labels.size(0))
end = time.time()

acc = correct/len(teset)

print(f'adapt test error {(1-acc)*100:.4f}')
print(f'The Time of code {(end-start):.4f}')
with open(f'../acc_result/diffusion_voting_noadaptnet_{args.dataset}_{args.corruption}_level={args.level}_{args.ifshuffle}.txt',"a") as f:
    f.write(f'{args}\n')
    f.write(f'Diffusion return highest adapt test accuracy {(np.mean(acc))*100:.2f}% Time:{(end-start):.4f}s\n')
****************************************************************************************************************************
I tried to reproduce it with the above code, but the test accuracy of ImageNet-C level5 gaussian_noise is only 0.1%.

**Comment by CCJing14** [comment]
`from __future__ import print_function

import argparse

import numpy as np
import torch
import torch.nn as nn
import torch.optim as optim
import torch.backends.cudnn as cudnn
cudnn.benchmark = True
from PIL import Image
from tqdm import tqdm
from utils.adapt_helpers import adapt_single, test_single, adapt_single_select,adapt_single_diffusion,test_ensemble,test_ensemble_debug,test_highest,test_voting
from utils.third_party import aug
from utils.train_helpers import build_model, prepare_test_data, prepare_test_imagenet100, tr_transforms, te_transforms, te_transforms_inc, common_corruptions
import torch
from torch.utils.data import DataLoader, Dataset
import numpy as np
import models.Res as Resnet
import time
import os
from pipeline.deccoruptor_lcm_pipe import IP2PLatentConsistencyModelPipeline
from transformers import CLIPTokenizer, CLIPTextModel, CLIPImageProcessor
from diffusers import AutoencoderKL, UNet2DConditionModel, LCMScheduler
from PIL import Image, ImageOps
import torchvision.transforms as transforms
import torchvision
from glob import glob 
from natsort import natsorted 

parser = argparse.ArgumentParser()
parser.add_argument('--dataroot', default="/workspace/")
parser.add_argument('--use_rvt', action='store_true')
parser.add_argument('--use_resnext', action='store_true')
parser.add_argument('--level', default=5, type=int)
parser.add_argument('--corruption', default='gaussian_noise')
parser.add_argument('--dataset', default='imagenet')
parser.add_argument('--resume',  default="../../imagenet-exps/results/imagenet_rn50")
parser.add_argument('--batch_size', default=1, type=int)
parser.add_argument('--prior_strength', default=16, type=int)
parser.add_argument('--optimizer', default='sgd')
parser.add_argument('--lr', default=0.00025, type=float)
parser.add_argument('--weight_decay', default=5e-4, type=float)
parser.add_argument('--niter', default=1, type=int)
parser.add_argument('--ifshuffle',  action='store_true', help='Set flag to True')
parser.add_argument('--workers', default=8, type=int, metavar='N',help='number of data loading workers (default: 4 )')
parser.add_argument('--method', default='adapt_single_selects_1_3',help='adapt_single, adapt_single_selects')
parser.add_argument('--ifdebug',  action='store_true', help='Set flag to True')
args = parser.parse_args()

net = build_model(args)

if args.dataset == 'imagenet':
    teset, testloader = prepare_test_data(args, use_transforms=False)
if args.dataset == 'imagenet100':
    teset, _ = prepare_test_imagenet100 (args)

print(f'Resuming from {args.resume}...')
ckpt = torch.load(f'{args.resume}/ckpt.pth')

print('Running...')
correct = 0

total = 0.0
start = time.time()
model_id = "/root/.cache/huggingface/hub/Anonymous-12/DeCorruptor-CM"
scheduler = LCMScheduler.from_pretrained(model_id, subfolder="scheduler")
pipe = IP2PLatentConsistencyModelPipeline.from_pretrained(model_id,
                                            torch_dtype=torch.float16, 
                                            scheduler=scheduler,
                                            use_safetensors=True, 
                                            output_type="tensor",
                                            safety_checker=None)
pipe = pipe.to("cuda")

generator = torch.Generator(device="cuda").manual_seed(1)
idx=0
for data in tqdm(testloader):
    idx += 1

    images, labels = data
    images, labels = images.cuda(), labels.cuda()

    out_image = pipe(prompt=['Clean the image'], 
            image=[images],
            num_images_per_prompt=1,
            num_inference_steps=4, 
            generator=generator,
            output_type="tensor",
            guidance_scale=7.5).images
    inputs = torch.tensor(out_image).permute(0, 3, 1, 2).cuda()

    if args.ifdebug==True:
        cor, conf, ent, pred = test_ensemble(net, inputs, labels, args.corruption, args.prior_strength,args.ifdebug,idx)
    cor, conf, ent, pred = test_ensemble(net,images, labels, args.corruption, args.prior_strength,args.ifdebug)
    
    correct += cor
    
    total += float(labels.size(0))
end = time.time()

acc = correct/len(teset)

print(f'adapt test error {(1-acc)*100:.4f}')
print(f'The Time of code {(end-start):.4f}')
with open(f'../acc_result/diffusion_voting_noadaptnet_{args.dataset}_{args.corruption}_level={args.level}_{args.ifshuffle}.txt',"a") as f:
    f.write(f'{args}\n')
    f.write(f'Diffusion return highest adapt test accuracy {(np.mean(acc))*100:.2f}% Time:{(end-start):.4f}s\n')`


I tried to reproduce it with the above code, but the test accuracy of ImageNet-C level5 gaussian_noise is only 0.1%.

**Comment by oyt9306** [comment]
Thank you for your interest in our work!

Unfortunately, we no longer have the specific code used for the table you mentioned — apologies for the inconvenience.

From what I remember, the ensemble was not performed in an online, on-the-fly manner. Instead, when using a real (corrupted) image for inference, we pre-generated and saved multiple reverted images offline (equal to the number of ensemble samples) and used it for ensemble at test-time. During test time, the real image was combined with these pre-generated ensemble samples for evaluation.

You can refer to the ensemble code provided at https://github.com/shiyegao/DDA for implementation.

Feel free to reach out if you have any further questions.
